lib/govuk_publishing_components/presenters/contextual_navigation.rb

```diff
@@ -6,9 +6,10 @@ class ContextualNavigation
 
       # @param content_item A content item hash with strings as keys
       # @param request_path `request.path`
-      def initialize(content_item, request_path)
+      def initialize(content_item, request_path, active_step_nav_content_id = false)
         @content_item = content_item
         @request_path = simple_smart_answer? ? content_item['base_path'] : request_path
+        @active_step_nav_content_id = active_step_nav_content_id
```

Comment:
As mentioned in the view, If you changed the definition of the constructor to:
`def initialize(content_item, request)`

Then you don't need to pass in the `active_step_nav_content_id` with a default value. Also, `false` is misleading, the default value should be `nil` as an actual value (not `true`) is being passed in.

You could use `request` to assign the rest of the values:
```
@request_path = simple_smart_answer? ? content_item['base_path'] : request.path
@active_step_nav_content_id =  request.query_parameters ? request.query_parameters['step-by-step-nav'] : nil
```